src/libstore-test-support/test-main.cc

```diff
@@ -0,0 +1,44 @@
+#include <cstdlib>
+
+#include "nix/store/globals.hh"
+#include "nix/util/logging.hh"
+
+#include "nix/store/tests/test-main.hh"
+
+namespace nix {
+
+int testMainForBuidingPre(int argc, char ** argv)
```

Comment:
I thought this was already factored out previously, but it wasn't.
Nonetheless, this should also be called by `libexpr-tests/main.cc`.